go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -292,74 +279,44 @@ func (npw *nodePortWatcher) createLbAndExternalSvcFlows(service *kapi.Service, s
-				fmt.Sprintf("cookie=%s, priority=110, in_port=%s, %s, %s=%s, tp_dst=%d, actions=ct(commit,zone=%d,nat(dst=%s:%s),table=6)",
-					cookie, npw.ofportPhys, flowProtocol, nwDst, externalIPOrLBIngressIP, svcPort.Port, HostNodePortCTZone, npw.gatewayIPv4, svcPort.TargetPort.String()))
-		}
-		nodeportFlows = append(nodeportFlows,
-			// table 6, Sends the packet to the host
-			fmt.Sprintf("cookie=%s, priority=110, table=6, actions=output:LOCAL",
-				etpSvcOpenFlowCookie),
-			// table 0, Matches on return traffic, i.e traffic coming from the host networked pod's port, and unDNATs
-			fmt.Sprintf("cookie=%s, priority=110, in_port=LOCAL, %s, tp_src=%s, actions=ct(commit,zone=%d nat,table=7)",
-				cookie, flowProtocol, svcPort.TargetPort.String(), HostNodePortCTZone),
-			// table 7, the packet back out eth0 to the external client
-			fmt.Sprintf("cookie=%s, priority=110, table=7, "+
-				"actions=output:%s", etpSvcOpenFlowCookie, npw.ofportPhys))
-
-		npw.ofm.updateFlowCacheEntry(key, nodeportFlows)
-
-	} else if isServiceTypeETPLocal && !hasLocalHostNetworkEp && config.Gateway.Mode == config.GatewayModeLocal {
+		externalIPFlows = append(externalIPFlows, npw.createFlowsForLocalHostNetEp(cookie, svcPort, flowProtocol, externalIPOrLBIngressIP, nwDst, nwSrc)...)
+	} else if config.Gateway.Mode == config.GatewayModeShared {
 		// case2 (see function description for details)
-		var nodeportFlows []string
-		klog.V(5).Infof("Adding flows on breth0 for %s Service %s in Namespace: %s since ExternalTrafficPolicy=local", ipType, service.Name, service.Namespace)
-		nodeportFlows = npw.createFlowsForNonLocalHostNetEp(cookie, svcPort.Port, flowProtocol, externalIPOrLBIngressIP, nwDst, nwSrc)
-		npw.ofm.updateFlowCacheEntry(key, nodeportFlows)
-	} else {
-		// case3 (see function description for details)
-		npw.ofm.updateFlowCacheEntry(key, []string{
+		externalIPFlows = append(externalIPFlows,
 			fmt.Sprintf("cookie=%s, priority=110, in_port=%s, %s, %s=%s, tp_dst=%d, "+
 				"actions=%s",
 				cookie, npw.ofportPhys, flowProtocol, nwDst, externalIPOrLBIngressIP, svcPort.Port, actions),
 			fmt.Sprintf("cookie=%s, priority=110, in_port=%s, %s, %s=%s, tp_src=%d, "+
 				"actions=output:%s",
-				cookie, npw.ofportPatch, flowProtocol, nwSrc, externalIPOrLBIngressIP, svcPort.Port, npw.ofportPhys),
-			npw.generateArpBypassFlow(protocol, externalIPOrLBIngressIP, cookie)})
+				cookie, npw.ofportPatch, flowProtocol, nwSrc, externalIPOrLBIngressIP, svcPort.Port, npw.ofportPhys))
 	}
+	npw.ofm.updateFlowCacheEntry(key, externalIPFlows)
 
 	return nil
 }
 
 // generate ARP/NS bypass flow which will send the ARP/NS request everywhere *but* to OVN
 // OpenFlow will not do hairpin switching, so we can safely add the origin port to the list of ports, too
-func (npw *nodePortWatcher) generateArpBypassFlow(protocol string, ipAddr string, cookie string) string {
+func (npw *nodePortWatcher) generateArpBypassFlow(protocol string, ipAddr string, cookie string) []string {
```

Comment:
why is this changed to return a slice when the logic of the function hasn't changed to return more than 1 value?

Reply:
haha ok my bad, I was lazy so changed return value to hack my need to append the results. let me change this back and do the needful at the right place in code.